### PR TITLE
feat: Enhance InstalledProductIDs to support more lines/filters

### PR DESCRIPTION
**Pull request by huali027**
* Add file delimiter and file path to the filters to keep the mapping between data and file

### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [x] Is this PR an enhancement?

### CODE REVIEW

**Comment by huali027** [comment]
@xiangce @JoySnow Please help to review, thanks.

File: insights/parsers/installed_product_ids.py [hunk 1]
```diff
@@ -88,23 +88,39 @@ class InstalledProductIDs(CommandParser):
             Brand Type:
             Brand Name:
 
-    Filters have been added to the parser so that only the ``ID`` element will
-    be collected.
+    Filters have been added to the parser so that only the filtered lines will be collected.
 
     Attributes:
         ids (set): set of strings of the unique IDs collected by the command
+        data(list): list of dicts of the attributes and its value
```
Comment (xiangce) [inline insights/parsers/installed_product_ids.py:95]: `data` is not a good name anymore to use as attributes, as itself cannot give any hit about what kind of the `data` is. Please rename the attribute more meaningful, e.g., `product_list`, etc.

File: insights/parsers/installed_product_ids.py [hunk 2]
```diff
@@ -88,23 +88,39 @@ class InstalledProductIDs(CommandParser):
...
+                one_file_data = {}
+                valid_data = True
+            else:
+                if valid_data:
+                    name, value = line.split(':', 1)
+                    one_file_data[name.strip()] = value.strip()
+        if one_file_data:
+            # add the last file data
+            self.data.append(one_file_data)
+        self.ids = set([item['ID'] for item in self.data])
```
Comment (xiangce) [inline insights/parsers/installed_product_ids.py:126]: the code can be simplified a bit as:
```python
        one_file_data = None
        for line in content:
            line = line.strip()
            # different file delimiter
            if line == 'Product Certificate':
                one_file_data ={}
                self.data.append(one_file_data)
            elif ":" in line and one_file_data is not None:
                    name, value = line.split(':', 1)
                    one_file_data[name.strip()] = value.strip()

        self.ids = set([item['ID'] for item in self.data])


```

**Review by xiangce — CHANGES_REQUESTED** [review]

File: insights/parsers/installed_product_ids.py (hunk 1 above)
Comment (huali027) [reply insights/parsers/installed_product_ids.py:95]: @xiangce sure, I'll rename it.

File: insights/parsers/installed_product_ids.py (hunk 2 above)
Comment (huali027) [reply insights/parsers/installed_product_ids.py:126]: @xiangce  yes, it is simple. The only concern is:
```
one_file_data ={}
self.data.append(one_file_data)
```
It adds an empty dict first. So I'm not sure if there is any case that there are no filtered data after it. But maybe we don't need to consider too much now?

File: insights/parsers/installed_product_ids.py [hunk 3]
```diff
@@ -13,7 +13,7 @@
 from insights.specs import Specs
 from .. import parser, CommandParser
 
-add_filter(Specs.subscription_manager_installed_product_ids, 'ID:')
+add_filter(Specs.subscription_manager_installed_product_ids, ['ID:', 'Product Certificate', 'Path'])
```
Comment (JoySnow) [inline insights/parsers/installed_product_ids.py:16]: IIUC, the added `'Path'` is used to filter the lines like `Path: /etc/pki/product-default/69.pem` .  If this is the case, what about adjusting the filter string to `'Path: '` to gain more accuracy?

File: insights/parsers/installed_product_ids.py [hunk 4]
```diff
@@ -88,23 +88,37 @@ class InstalledProductIDs(CommandParser):
...
         for line in content:
-            if line.strip().startswith('ID:'):
-                _, id = line.strip().split(':', 1)
-                self.ids.add(id.strip())
+            # different file delimiter
+            if line == 'Product Certificate':
+                if one_file_data:
+                    self.product_certs.append(one_file_data)
+                one_file_data = {}
+            else:
```
Comment (JoySnow) [inline insights/parsers/installed_product_ids.py:117]: When there are no ":" in `line`, this following `split()` will run into exception, eg: 
```
>>> name, value = "some line".split(':', 1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: not enough values to unpack (expected 2, got 1)
```

What about adding the checking of  `line` inside this `else` clause ?
```suggestion
            elif line and ":" in line:
```

File: insights/parsers/installed_product_ids.py [hunk 5]
```diff
@@ -88,23 +88,37 @@ class InstalledProductIDs(CommandParser):
...
+                if one_file_data is not None:
+                    name, value = line.split(':', 1)
+                    one_file_data[name.strip()] = value.strip()
```
Comment (JoySnow) [inline insights/parsers/installed_product_ids.py:120]: For the key variables naming, to keep a better consistency, what about converting the key to the lowercase and joined the words with low dash ? 
 
For example, put the key variables name to: 
```
    assert results.product_certs[1]['Path'] == '/etc/pki/product/479.pem'.    ----> 'path'
    assert results.product_certs[1]['ID'] == '479'                            ----> 'id'
    assert results.product_certs[1]['Brand Type'] == ''                       ----> 'brand_type'
```

**Review by JoySnow — CHANGES_REQUESTED** [review]

File: insights/parsers/installed_product_ids.py (hunk 3 above)
Comment (huali027) [reply insights/parsers/installed_product_ids.py:16]: sure, I'll update it.

File: insights/parsers/installed_product_ids.py (hunk 5 above)
Comment (huali027) [reply insights/parsers/installed_product_ids.py:120]: yes, good to me.

**Review by xiangce — APPROVED** [review]

**Pull request merged**